### ISSUE TITLE
Installation instructions are incorrect

### ISSUE THREAD

**Issue by wwat** [HoTT/HoTT#729]
You refer to URL https://github.com/coq/coq/branches/trunk which returns a HTTP code 410.

**Comment by wwat** [comment]
Thanks for updating the instructions. Unfortunately, they are still incorrect, or the code is incorrect. 
```
HOQC theories/Types/Sigma
File "./theories/Types/Sigma.v", line 294, characters 0-255:
Anomaly: Uncaught exception Retyping.RetypeError(3). Please report.
make: *** [theories/Types/Sigma.vo] Error 1
```
Assuming that you are an expert on this stuff, would you (or someone reading this) be so kind to point me at some intuition/very lengthy explanation of why induction over an `eq` makes sense and/or is useful. I have no issues with structural induction, or even induction over transfinite ordinals, etc., but I lack any kind of intuition for induction over `eq`. 

Perhaps some drawing of the structure would actually make sense, because it cannot possibly be that complicated. 

I understand that it is useful conceptually to provide algorithmic descriptions to relate objects, but I have never seen applications of identifications beyond `eq_refl`, which makes it kind of a hard thing to learn; learning in the full abstract rarely works (all one would have is a collection of facts/rules, without a broader understanding of the concepts). I have spent some semesters in modal logic, if that helps. 

For example, in modal logic, there is the K axiom, but somehow it doesn't appear to be the same one that you are using (see http://en.wikipedia.org/wiki/Modal_logic ).

**Comment by spitters** [comment]
Quoting @JasonGross 

> That sounds like you are using .vo files compiled with a different version
of Coq, so, yes, make clean, then make. Or `find theories coq -name "*.vo"
| xargs rm`.

Regarding path induction, we wrote a book, which is freely available:
http://homotopytypetheory.org/book/

The K-axiom in modal logic is different. If you have not studied these issues, I am not sure that this library would be very useful for you.

**Comment by wwat** [comment]
I attempted to read the book,  but I consider that it is written using fairly abstract and ambiguous language (like reusing symbols in various contexts, heavy context depending overloading) and generally assuming a very high level of mathematical sophistication. While I can sort of understand that you didn't want to write 1500 pages, I don't think it's a very accessible text. 

If the goal is to convince the general mathematician (let's take that to be the 1% best of the population), then I think that failed. If the goal was to get the top 0.01% to understand it, you might have succeeded. 

In the case of "K", it's rather stupid for two branches of mathematics to both claim that name and if one then refers to it, not mention that it has nothing to do with the other. In a book written by supposedly the best humanity has to offer on mathematics, it's still a grave error to not be aware of that. At least, that's my opinion. That being said, you can easily add a footnote now saying that it has absolutely nothing to do with that other axiom "K". If this book has had any proofreading done on it, that should already have happened, imho. 

Did you ever do a survey on how many people actually completed the book? (That is, of people who are not your direct students or colleagues, obviously.) If you did not do this, how can you evaluate whether your research had any success (which you have claimed in various sources)? 

You should not interpret the above as useless whining, but instead, I would like you to actually do something with this feedback.

**Comment by mikeshulman** [comment]
We did not do a numerical survey (I'm not even sure how that would be possible), but we have received positive feedback from a large number of mathematicians.  Most mathematicians do not do a numerical survey of how many people read their work (do any?), but nevertheless feel able to say that it was a success.

It also happens very often in mathematics that the same names and letters are used for different things in different contexts.  I don't think there is much danger of confusion about axiom K because we specifically say when we introduce it what it means, so that it should be clear to the reader (as indeed it was clear to you, from your comment) that it is unrelated to the modal logic one.  If you'd like to point out specific instances of overloading in the book that you think are worse than the general norm in mathematics, you can open separate issues for them; I'm closing this issue because its original subject (the installation instructions) was fixed by dd670da.

**Comment by JasonGross** [comment]
The retyping anomaly is actually [a bug in the new version of Coq](https://coq.inria.fr/bugs/show_bug.cgi?id=4097); you should be using the pinned version.

**Comment by andrejbauer** [comment]
There are other introductions to homotopy type theory out there, written by various people. Maybe some of those can get you started. Here's a couple:

* Bob Harper's course (with video lectures): http://www.cs.cmu.edu/~rwh/courses/hott/
* François G. Dorais has a series of posts, and since he's a relative "outsider" those might actually be more understandable: http://logic.dorais.org (see "Recent Posts" at the bottom).

**Comment by wwat** [comment]
Thanks for all of the links.

Those lectures seem useful (judging by the titles), but can one also download those videos for offline use? 

I don't particularly get the point of making it harder to download content myself (I can obviously capture the video and audiochannels in parallel and rip the content, if I had more time than I have today) and I am sure someone else has already done that.  (I understand that you are probably not affiliated with those people.)

**Comment by andrejbauer** [comment]
I am somewhat affiliated with "those people" but I am definitely not affiliated with Silverlight :-)
